sonar-plugin-api/src/main/java/org/sonar/api/internal/ApiVersion.java

```diff
@@ -23,29 +23,27 @@
 import java.io.IOException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
-import javax.annotation.Nullable;
-import org.sonar.api.SonarProduct;
-import org.sonar.api.SonarQubeSide;
-import org.sonar.api.SonarQubeVersion;
 import org.sonar.api.utils.System2;
 import org.sonar.api.utils.Version;
 
 /**
- * For internal use only.
+ * For internal use
+ *
+ * @since 6.0
  */
-public class SonarRuntimeFactory {
+public class ApiVersion {
 
   private static final String FILE_PATH = "/sq-version.txt";
 
-  private SonarRuntimeFactory() {
-    // prevents instantiation
+  private ApiVersion() {
+    // only static methods
   }
 
-  public static SonarQubeVersion create(System2 system, SonarProduct product, @Nullable SonarQubeSide sonarQubeSide) {
+  public static Version load(System2 system) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)